formats/vector-to-label-plugin/src/polus/plugins/formats/vector_to_label/__main__.py

```diff
@@ -0,0 +1,325 @@
+            future_x.squeeze(),  # type: ignore[union-attr]
+            current_x,
+            labels,
+        )
+    if z > 0:
+        labels, labels_z, indices_z = reconcile_overlap(
+            future_z.squeeze(),  # type: ignore[union-attr]
+            current_z,
+            labels,
+        )
+
+    uniques, labels = numpy.unique(labels, return_inverse=True)
+    labels = numpy.asarray(labels, numpy.uint32)
+    labels[labels > 0] = labels[labels > 0] + offset
+    max_label = numpy.max(uniques) + offset
+
+    if y > 0:
+        # noinspection PyUnboundLocalVariable
+        for label, index in zip(labels_y, indices_y):
+            if index.size == 0:
+                continue
+            labels[index] = label
+
+    if x > 0:
+        # noinspection PyUnboundLocalVariable
+        for label, index in zip(labels_x, indices_x):
+            if index.size == 0:
+                continue
+            labels[index] = label
+
+    if z > 0:
+        # noinspection PyUnboundLocalVariable
+        for label, index in zip(labels_z, indices_z):
+            if index.size == 0:
+                continue
+            labels[index] = label
+
+    # Zarr axes ordering should be (t, c, z, y, x). Add missing t, c, and z axes
+    labels = numpy.asarray(numpy.reshape(labels, shape), dtype=numpy.uint32)
+    if ndims == 2:  # noqa: PLR2004
+        labels = labels[numpy.newaxis, numpy.newaxis, numpy.newaxis, :, :]
+    else:
+        labels = labels[numpy.newaxis, numpy.newaxis, :, :, :]
+
+    # noinspection PyTypeChecker
+    zarr_root = zarr.open(str(zarr_path))[0]
+    zarr_root[0:1, 0:1, z_min:z_max, y_min:y_max, x_min:x_max] = labels
+
+    if ndims == 2:  # noqa: PLR2004
+        return None, labels[0, 0, 0, -1, :], labels[0, 0, 0, :, -1], max_label
+
+    return (
+        labels[0, 0, -1, :, :],
+        labels[0, 0, :, -1, :],
+        labels[0, 0, :, :, -1],
+        max_label,
+    )
+
+
+def convert_file(
+    inp_path: pathlib.Path,
+    out_dir: pathlib.Path,
+) -> None:
+    """Converts a single '_flows.ome.zarr' file to a label file.
+
+    Args:
+        inp_path: Path to the '_flows.ome.zarr' file.
+        out_dir: Path to the output directory.
+    """
+    with bfio.BioReader(inp_path) as reader:
+        reader_shape = (reader.Z, reader.Y, reader.X)
+        metadata = reader.metadata
+
+    zarr_path = out_dir.joinpath(
+        l2v_helpers.replace_extension(inp_path, extension="_tmp.ome.zarr"),
+    )
+    helpers.init_zarr_file(zarr_path, metadata)
+
+    threads: dict[
+        tuple[int, int, int],
+        concurrent.futures.Future[ThreadFuture],
+    ] = {}
+    thread_kwargs: dict[str, typing.Any] = {
+        "in_path": inp_path,
+        "zarr_path": zarr_path,
+        "coordinates": (0, 0, 0),
+        "reader_shape": reader_shape,
+        "future_z": None,
+        "future_y": None,
+        "future_x": None,
+    }
+
+    with concurrent.futures.ThreadPoolExecutor(
```

Comment:
this approach of creating dependencies is ok and probably does not really matter if we use a thread pool. However it is not immediately clear why we cannot do the convert of all tiles in parallel and in a second step merge results (for merging we do have some dependencies ). I would imagine this algorithm is compute bound and would benefit from more parallelization but maybe becnhmarks have shown otherwise.